### DISCUSSION TITLE
OpenHasp-HA-pyscript

### DISCUSSION THREAD

**Discussion by vmfs1968** [FredericG-BE/OpenHasp-HA-pyscript#14] [General]
hello
I already managed to get it working
good job.
The clock stops, is that how it is or is there something wrong?

**Comment by FredericG-BE** [comment]
Hi,
The clock stops in the helloWorld? 
No, that should not happen. Do you see anything special on the display serial output or in home-assistant.log in HA? 
If there is nothing special you could open mqtt explorer and verify that HA sends updates to the screen every minute (to change the clock) and you could verify that the screen periodically sends a discovery message (the latter would show that the screen is up and still connected to the mqtt broker)
Hope this helps

**Comment by vmfs1968** [comment]
![image](https://github.com/FredericG-BE/OpenHasp-HA-pyscript/assets/41055800/1ae3f1d3-dafe-4e47-9b8e-8e52e4964cd1)
do not send every minute
there is no where the problem is

![image](https://github.com/FredericG-BE/OpenHasp-HA-pyscript/assets/41055800/d9aa9550-aead-474c-af7d-ffab34a2ab6a)
If I reload the integration it updates the clock

This is the configuration I put in

pyscript:
  allow_all_imports: true
  hass_is_global: true

apps:

# openhasp_helloWorld: 

  openhasp_demo:
    - friendly_name: "Bureau"
      plate_name: "plate_test1"
      resolution_x: 480
      resolution_y: 320
      mediaplayer: "media_player.bureau"
      lamp: "light.bureau_spots"
      camera: "camera.oost" # This is optional

**Comment by FredericG-BE** [comment]
The clock never gets any updates? 
Perhaps this is the same as https://github.com/FredericG-BE/OpenHasp-HA-pyscript/issues/13 ?
You could have a look at the advise I gave there. I will ask to elaborate how he fixed it, is not very clear to me what the issue was in his case

Thanks

**Comment by vmfs1968** [comment]
via telnet this appears + - every 5 minutes.
watch does not update

#[12:23:06.788][86004/102900 16][33048/33660  2] MQTT PUB: statusupdate => {"node":"plate35","idle":"short","version":"0.7.0-rc12","uptime":5726,"ssid":"Domotica","rssi":-71,"ip":"192.168.68.154","mac":"3C:84:27:C1:0C:3C","heapFree":104492,"heapFrag":17,"core":"4.4.6","canUpdate":"false","page":2,"numPages":12,"tftDriver":"NV3041A","tftWidth":480,"tftHeight":272}
#[12:23:06.807][86004/104492 17][33048/33660  2] MQTT PUB: sensors => {"time":"2024-05-30T12:23:06","uptimeSec":5726,"uptime":"0T01:35:26"}

**Comment by FredericG-BE** [comment]
I would:
- Make sure the default openHASP integration is not active (while I am unsure if this could cause issues when it is not configured to talk to the plate)
- in your configuration (yaml) only enable the openhasp_helloWorld and not the openhasp_demo under "apps:" (I suppose you are not using that right now) and restart pyscript
- check home-assistant.log for pyscript warnings or errors. You could increase the level of logging details as described in https://github.com/FredericG-BE/OpenHasp-HA-pyscript/issues/13

**Comment by vmfs1968** [comment]
it already works. Thank you very much for your attention. in my case it was, I disabled the openhasp.yaml file and it now works. 
I would like to have seconds on the watch, can you put that in?

**Reply by FredericG-BE** [reply]
Good that it works. I will add do the readme to disable the openhasp.yaml

The red hand you see in some of the pictures indicate alarm-time. 
I could add another hand for seconds if you would like that; I created https://github.com/FredericG-BE/OpenHasp-HA-pyscript/issues/15

**Comment by vmfs1968** [comment]
Yes, I would really like to have this function on my watch

**Comment by FredericG-BE** [comment]
I added the second hand, just a few lines of code... https://github.com/FredericG-BE/OpenHasp-HA-pyscript/commit/b81169abecd9703e5ad005e3a290be5596282dba
The first minute the second indicator will not show the correct seconds (as it always starts from 0) but it will synchronize

It is enabled by setting the showSec parameter in the AlarmClock to True, it is now also enabled in the helloWorld

https://github.com/FredericG-BE/OpenHasp-HA-pyscript/assets/11998085/073a8d84-8b15-4b8a-9cd4-ced263769caf

**Comment by vmfs1968** [comment]
it's working fine.
One of the things I would also like to have was the weather forecast

**Reply by FredericG-BE** [reply]
I have no experience regarding weather forecast data in HA. Any design you have in mind? 
To be honest, not too interested myself to work on that right now... I can however assist if needed
